### PR TITLE
[Test] Add ParamTest scaffolding for model tests 

### PR DESCRIPTION
**Pull request by zhoonit**
Add Parameterized test scaffolding for integrated tests with minor
changes

**Additional Changes proposed in this PR:**
- Add `TensorDim::TensorDim(const std::string &shape)
- Move `Tensor::epsilon` to public

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>
cc: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: fix_app.patch [hunk 1]
```diff
@@ -1,35 +0,0 @@
-diff --git a/Applications/TransferLearning/Draw_Classification/jni/main.cpp b/Applications/TransferLearning/Draw_Classification/jni/main.cpp
```
Comment (zhoonit) [inline fix_app.patch:1]: Not related to the PR, There was a garbage file hiding in the project folder

File: nntrainer/include/tensor_dim.h [hunk 2]
```diff
@@ -45,6 +45,8 @@ class TensorDim {
 
   TensorDim(const TensorDim &rhs) = default;
 
+  TensorDim(const std::string &shape) : TensorDim() { setTensorDim(shape); }
```
Comment (kparichay) [inline nntrainer/include/tensor_dim.h:48]: setTensorDim can fail with error return code.
please check that.

Comment (zhoonit) [reply nntrainer/include/tensor_dim.h]: Done

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**